### ISSUE TITLE
Correspondence Establishing

### ISSUE THREAD

**Issue by Pamphlett** [MIT-SPARK/TEASER-plusplus#38]
Hi,

Thanks for sharing your code!
I'm wondering about the role (or significance) of the putative correspondence that fed to the whole algorithm. 
Why do we need to use a set of more reliable correspondences (e.g. computed by FPFH) instead of some heuristic (or naive) methods (e.g. nearest neighbor)?

I'm interested in the correspondence-free case you presented in Fig.1.(g). As the computation of FPFH consumes a considerable amount of computational resources which will probably hinder the real-time performance of the whole system.

Thx

**Comment by jingnanshi** [comment]
Hi @Pamphlett, thank you for your interests! Certainly you can use heuristics to generate correspondences, but if the two point clouds have zero overlap, then it's unlikely to have any inlier correspondences, and the resulted registration solution will be incorrect. 

The correspondence-free case means that all-to-all correspondences are generated and TEASER++ is used to filter out the incorrect ones. This will 1) consume lots of memory 2) be slower than the case if reliable correspondences are fed to TEASER++ instead.

Regarding the computation cost, I agree that FPFH computation is slow. There has been significant progress on learned 3D feature descriptors recently, see e.g. [FCGF](https://github.com/chrischoy/FCGF) and [3DSmoothNet](https://github.com/zgojcic/3DSmoothNet). In particular, FCGF is really fast and might be suitable for real-time applications. In addition, to further reduce computation time, you can downsample the point clouds then compute features.

So to summarize:
- If putative correspondences are somewhat good, it narrows down the solution space TEASER++ needs to consider, thus TEASER++ should run faster.
- Although TEASER++ can work on correspondence-free problems, it will consume a lot of memory and not suitable for real-time applications (unless the point clouds are small).
- Instead of using FPFH, use more recently developed learned feature descriptors instead.

**Comment by Pamphlett** [comment]
@jingnanshi I got it! Thanks for your clear answer ;)

**Comment by jingnanshi** [comment]
Glad I can help @Pamphlett
